docs/csharp/whats-new/tutorials/static-abstract-interface-methods.md

```diff
@@ -0,0 +1,151 @@
+
+:::code language="csharp" source="./snippets/staticinterfaces/Program.cs" id="TestAddition":::
+
+You can make this code more reusable by declaring that these types implement the appropriate arithmetic interfaces. The first change to make is to declare that `Point<T, T>` implements the `IAdditionOperators<Point<T>, Translation<T>, Point<T>>` interface. The `Point` type makes use of different types for operands and the result. The `Point` type already implements an `operator +` with that signature, so adding the interface to the declaration is all you need:
+
+```csharp
+public record Point<T>(T X, T Y) : IAdditionOperators<Point<T>, Translation<T>, Point<T>>
+    where T : IAdditionOperators<T, T, T>
+```
+
+Finally, when you're performing addition, it's useful to have a property that defines the additive identity value for that type. There's a new experimental interface for that feature: `IAdditiveIdentity<TSelf, TResult>`. A translation of `{0, 0}` is the additive identity: The resulting point is the same as the left operand. The `IAdditiveIdentity<TSelf, TResult>` interface defines one readonly property: `AdditiveIdentity`, that returns the identity value. The `Translation<T>` needs a few changes to implement this interface:
+
+:::code language="csharp" source="./snippets/staticinterfaces/Translation.cs":::
+
+There are a few changes here, so let's walk through them one by one. First, you declare that the `Translation` type implements the `IAdditiveIdentity` interface:
+
+```csharp
+public record Translation<T>(T XOffset, T YOffset) : IAdditiveIdentity<Translation<T>, Translation<T>> 
+```
+
+You next might try implementing the interface member as shown in the following code:
+
+```csharp
+public static Translation<T> AdditiveIdentity =>
+    new Translation<T>(XOffset: 0, YOffset: 0);
+```
+
+The preceding code won't compile, because `0` depends on the type. The answer: Use `IAdditiveIdentity<T>.AdditiveIdentity` for `0`. That change means that your constraints must now include that `T` implements `IAdditiveIdentity<T>`. That results in the following implementation:
+
+:::code language="csharp" source="./snippets/staticinterfaces/Translation.cs" id="AdditiveIdentity":::
+
+Now that you've added that constraint on `Translation<T>`, you need to add the same constraint to `Point<T>`:
+
+:::code language="csharp" source="./snippets/staticinterfaces/Point.cs":::
+
+This sample has given you a look at how the interfaces for generic math compose. You wrote a method that relied on the `INumber<T>` interface so that method could be used with any numeric type. You built a class that relies on the addition interfaces to implement a type that only supports one mathematical operation. That type declares its support for those same interfaces so it can be composed in other ways. The algorithms are written using the most natural syntax of mathematical operators.
```

Comment:
nit: Should this be in checkmark syntax? Maybe I'm wrong 😁

```suggestion
This sample has given you a look at how the interfaces for generic math compose. You learned how to:

> [!div class="checklist"]
>
> * Write a method that relied on the `INumber<T>` interface so that method could be used with any numeric type.
> * Build a type that relies on the addition interfaces to implement a type that only supports one mathematical operation.
>   That type declares its support for those same interfaces so it can be composed in other ways. The algorithms are written using the most natural syntax of mathematical operators.
```